recipes-asteroid/qml-asteroid-config/qml-asteroid-config_1.0.bb

```diff
@@ -0,0 +1,31 @@
+SUMMARY = "Generates a machine specific config file for AsteroidOS"
+LICENSE = "GPLv3"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/GPL-3.0;md5=c79ff39f19dfec6d293b95dea7b07891"
+
+PACKAGE_ARCH = "${MACHINE_ARCH}"
+
+python do_generate_config () {
+    machine_vars = ['MACHINE_DISPLAY_FLAT_TIRE', 'MACHINE_DISPLAY_BORDER_GESTURE_WIDTH', 'MACHINE_DISPLAY_ROUND', 'MACHINE_HAS_WLAN', 'MACHINE_HAS_SPEAKER']
+
+    from configparser import ConfigParser
+    config = ConfigParser()
+    config.optionxform=str
+    config.add_section('General')
+    for v in machine_vars:
+        if d.getVar(v) is not None:
+            config.set('General', v[len("MACHINE_"):], str(d.getVar(v)))
+
+    with open(d.getVar('WORKDIR') + "/machine.conf", "w") as machine_conf:
+        config.write(machine_conf)
+}
+
+do_install_append() {
+    if [ -f ../machine.conf ]; then
+      install -d ${D}/etc/qml-asteroid/
+      cp ../machine.conf ${D}/etc/qml-asteroid/machine.conf
+    fi
+}
+
+addtask do_generate_config before do_install
+
+FILES_${PN} += "/etc/qml-asteroid/"
```

Comment:
I wonder whether it makes sense to have this under the qml-asteroid umbrella. In principle, any Asteroid component could be interested in this config file, for example if you have a shell script which does one thing or the other depending on whether the machine has MACHINE_HAS_SPEAKER or not, it would be absurd to have it depend on a qml-asteroid config. What do you think of elevating this as a global asteroid config file?

Reply:
Excellent point, will change to /etc/asteroid